### PR TITLE
feat(vitals): Add web vitals display on transaction summary

### PR DESCRIPTION
**Pull request by Zylphrex**
There is currently no way of viewing web vitals as a time series in the
performance product. This change exposes a new web vitals display mode in the
transaction summary that contains first paint, first contentful paint, largest
contentful paint and first input delay. Cumulative layout shift is excluded at
this time as it is not a duration measurement and thus does not belong on this
chart.

### CODE REVIEW

**Comment by markstory** [comment]
Should the tooltip values be formatted as durations instead of floats?
Screen Shot 2020-11-05 at 10 58 43 AM

File: src/sentry/static/sentry/app/views/performance/transactionSummary/vitalsChart.tsx [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+    } = this.props;
+
+    const unselectedSeries = location.query.unselectedSeries ?? [];
+    const unselectedMetrics = Array.isArray(unselectedSeries)
+      ? unselectedSeries
+      : [unselectedSeries];
+    const seriesSelection = unselectedMetrics.reduce((selection, metric) => {
+      selection[metric] = false;
+      return selection;
+    }, {});
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/performance/transactionSummary/vitalsChart.tsx:90]: We have similar blocks in a few charts, should we consider making a function that takes the location and returns the seriesSelection? Maybe something to do separately.

File: src/sentry/static/sentry/app/views/performance/transactionSummary/vitalsChart.tsx [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+                    <ErrorPanel>
+                      <IconWarning color="gray500" size="lg" />
+                    </ErrorPanel>
+                  );
+                }
+                const colors =
+                  (results && theme.charts.getColorPalette(results.length - 2)) || [];
+
+                // Create a list of series based on the order of the fields,
+                // We need to flip it at the end to ensure the series stack right.
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/performance/transactionSummary/vitalsChart.tsx:200]: Does the stacking matter that much as this is a line series and not an area series?

File: tests/js/spec/views/performance/transactionSummary.spec.jsx [hunk 3]
```diff
@@ -7,11 +7,11 @@ import {mountWithTheme} from 'sentry-test/enzyme';
...
-function initializeData({hasTrendsFeature} = {}) {
-  const features = ['discover-basic', 'performance-view'];
-  if (hasTrendsFeature) {
-    features.push('trends');
-  }
+function initializeData({features: additionalFeatures = []} = {}) {
+  const features = ['discover-basic', 'performance-view', ...additionalFeatures];
+  // if (hasTrendsFeature) {
+  //   features.push('trends');
+  // }
```
Comment (markstory) [inline tests/js/spec/views/performance/transactionSummary.spec.jsx:14]: Could delete instead.

**Review by dashed — APPROVED** [review]

**Review by wmak — COMMENTED** [review]
Are we set on the display dropdown order? I thought this would appear under Duration Breakdown.

Also should we disable (or hide) the item when vitals aren't available (like we do for the tab)

**Comment by Zylphrex** [comment]
> Are we set on the display dropdown order? I thought this would appear under Duration Breakdown.
> 
> Also should we disable (or hide) the item when vitals aren't available (like we do for the tab)

No, we can change the dropdown order. Why would you expect it to appear below `Duration Breakdown`? My thinking was that the list of options was short enough that you can see all of them at once, so placing it at the end (keeping the existing alphabetical order) was fine.

And yeah, disabling the option is a good idea. Perhaps we should even opt to disable over hiding the tab.

**Pull request merged**